If an unrelated syslog/audit message slips between a complain-mode 
"PERMITTING x access to /bin/foo" and the immediately following 
"changing_profile" hint, logprof/genprof will get confused and not 
realize that a potential domain change is occurring.  This fixes 
bugzilla #175421.

---
 utils/SubDomain.pm |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: trunk/utils/SubDomain.pm
===================================================================
--- trunk.orig/utils/SubDomain.pm
+++ trunk/utils/SubDomain.pm
@@ -1431,7 +1431,9 @@ sub do_logprof_pass {
       if($mode =~ /x/) {
         # we need to try to check if we're doing a domain transition this time
         if($sdmode eq "PERMITTING") {
-          $stuffed = <LOG>;
+          do {
+            $stuffed = <LOG>;
+          } until $stuffed =~ /AppArmor|audit/;
           if($stuffed =~ m/changing_profile/) {
             $domainchange = "change";
             $stuffed = undef;
